index.html

```diff
@@ -270,10 +270,448 @@
+```
+]
+---
+# React avec Redux, en pratique
+## 1. react-redux `connect`
+
+Dans les deux cas, l'usage de *ConnectedTabs* reste le meme:
+
+  ```javascript
+  ...
+  render() {
+    const tabs = [{
+      name: 'developers',
+      label: 'Developers! Developers! Developers!',
+    }];
+
+    return (
+      ...
+      <ConnectedTabs tabs={tabs} />
+    );
+  }
+  ```
+---
+# React avec Redux, en pratique
+## 1. react-redux `connect`
+--
+- Souscris un composant aux changements de `state`
+--
+- Execute les fonctions `mapStateToProps` et `mapDispatchToProps` (*) passees en arguments
+--
+  - a **chaque rendu initie par le composant parent**
+--
+  - a **chaque modification du `state`**
+
+---
+# React avec Redux, en pratique
+## 1. react-redux `connect`
+A l'exception de `mapDispatchToProps` qui n'est execute qu'une fois dans sa forme `mapDispatchToProps(dispatch)` vs `mapDispatchToProps(dispatch, props)`
+---
+# React avec Redux, en pratique
+## 1. react-redux `connect`
+
+Par defaut, un composant connecte n'est re-rendu par `connect` *uniquement* lorsque le merge des proprietes finales n'est pas shallowequal a ses props precedentes.
+
+Si vous etes frileux en ce qui concerne les `PureComponent` et que vous utilisez `connect`:
+
+_SURPRISE_: Vous utilisez _deja_ des `PureComponent`  - et, a priori pas qu'un peu ;)
+
+---
+# React avec Redux, en pratique
+## 2. Les selecteurs
+.smallest[
+```javascript
+  // {
+  //   tabs: [{
+  //     name: 'developers',
+  //     label: 'Developers, developers, developers'
+  //   }]
+  // }
+  const selectTabs = (state) => state.tabs;
+```
+]
+
+--
+.smallest[
+```javascript
+  const selectOneTab = (state, name) =>
+    selectTabs(state).find((tab) => tab.name === name);
+```
+]
+--
+.smallest[
+```javascript
+  const selectOneTranslatedTab = (state, name) => {
+    const tab = selectOneTab(state, name);
+    return tab
+      ? { ...tab, translated_label: translate(tab.name) }
+      : undefined
+  };
+```
+]
+---
+# React avec Redux, en pratique
+## 2. `createSelector`
+.smallest[
+```javascript
+  // {
+  //   tabs: [{
+  //     name: 'developers',
+  //     label: 'Developers, developers, developers'
+  //   }]
+  // }
+  const selectTabs = (state) => state.tabs;
+```
+]
+
+--
+.smallest[
+```javascript
+  const passName = (state, name) => name;
+  const selectOneTab = createSelector(
+    [selectTabs, passName],
+    (tabs, name) => tabs.find((tab) => tab.name === name),
+  );
+```
+]
+--
+.smallest[
+```javascript
+  const selectOneTranslatedTab = createSelector(
+    [selectOneTab, passName],
+    (tab) => tab && { ...tab, translated_label: translate(tab.name) },
+  );
+```
+]
+---
+# React avec Redux, en pratique
+## .smaller[2.1 Fonctions `reducer`: une approche naive]
+.smallest[
+```javascript
+function reducer(state, action) {
+  if (action.type === 'SET_ACTIVE_TAB') {
+    return {
+      tabs: state.tabs.map((tab) => ({
```

Comment:
Hop, let's go to the pub!

Wait a minute, something's wrong here.

Can someone in the audience help our young lad here see the problem and get to go to the pub?